app/Http/Controllers/TasksController.php

```diff
@@ -3,18 +3,33 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
+use Session;
+use \App\Task;
+
+
 
 class TasksController extends Controller
 {
+
+    
+    
     /**
      * Display a listing of the resource.
      *
      * @return \Illuminate\Http\Response
      */
     public function index()
+    {   
+        return $this->fetchTasks();
+    }
+
+    public function fetchUser()
     {
-        $tasks = \App\Task::all();
-        return $tasks;
+        return Session::get('username');
+    }
+
+    public function fetchTasks() {
+       return Task::where('created_by', $this->fetchUser())->get();
```

Comment:
See comment on `TaskController`. Also, if you're writing methods like this that are being reused by multiple controllers, it may be a good idea to create your own base controller that includes this functionality, make the methods `protected`, and instead of `extends Controller`, you would `extends MyCustomController`